docs/development/etcdoperatortask-design.md

```diff
@@ -0,0 +1,501 @@
+- The `config` field uses `runtime.RawExtension` to flexibly support task-specific parameters, validated via admission webhook.
+- `ownerEtcdReference` under the spec field is renamed to `etcdReference` for clarity.
+- The `LastOperation` struct now uses a `Description` field instead of `Name` and `Reason` for better expressiveness.
+
+---
+
+### Go API Definition
+
+<details>
+<summary>Show EtcdOperatorTask CRD Go Definition</summary>
+
+```go
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:path=etcdoperatortasks,shortName=eot;eots,scope=Namespaced
+// +kubebuilder:subresource:status
+type EtcdOperatorTask struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+
+    Spec   EtcdOperatorTaskSpec   `json:"spec"`
+    
+    Status EtcdOperatorTaskStatus `json:"status,omitempty"`
+}
+
+// +kubebuilder:validation:XPreserveUnknownFields
+// +kubebuilder:validation:Immutable
+type EtcdOperatorTaskSpec struct {
+    // +required
+    Type v1alpha1.EtcdOperatorTaskType `json:"type"`
+
+    // Config is task-specific key/value parameters.
+    // +required
+    Config runtime.RawExtension `json:"config,omitempty"` 
+
+    // TTLSecondsAfterFinished controls how long the status+pod stays around.
+    // +optional
+    // +kubebuilder:validation:XValidation:message="TTLSecondsAfterFinished must be greater than 0",rule="!has(self) || self > 0"
+    TTLSecondsAfterFinished *int32 `json:"ttlSecondsAfterFinished,omitempty"`
+
+    // +optional
+    EtcdReference types.NamespacedName `json:"etcdReference,omitempty"`
+
+}
+
+```
+</details>
+
+> **Best Practice:**
+> Use `runtime.RawExtension` for the `config` field to allow each task type to define its own schema. Validate this field with an admission webhook for type safety and user feedback.
+
+---
+
+### Status Subresource
+
+The `status` field tracks the progress and outcome of each task, including state transitions, errors, and operation history.
+
+<details>
+<summary>Show EtcdOperatorTaskStatus Go Definition</summary>
+
+```go
+// EtcdOperatorTaskStatus is the status for a EtcdOperatorTask resource.
+type EtcdOperatorTaskStatus struct {
+  // State is the last known state of the task.
+  State TaskState `json:"state"`
+  // Time at which the task has moved from "pending" state to any other state.
+  // +optional
+  InitiatedAt *metav1.Time `json:"initiatedAt"`
+  // LastError represents the errors when processing the task. Will have a limit of 10 entries at a time.
+  // +optional
+  LastErrors []LastError `json:"lastErrors,omitempty"`
+  // Captures the last operation status if task involves many stages.
+  // +optional
+  LastOperation *LastOperation `json:"lastOperation,omitempty"`
+}
+
+type LastOperation struct {
+  // Status of the last operation, one of pending, progress, completed, failed.
+  State OperationState `json:"state"`
```

Comment:
How is this different from `Status.State`? I see an overlap of information here.